### PR TITLE
Added http/socks proxy support

### PR DESCRIPTION
**Pull request by lins05**
~Just tested it manually, I'll add an integration test soon.~

Integration test is added now.

### CODE REVIEW

File: src/clj/pod/epiccastle/bbssh/pod/session.clj [hunk 1]
```diff
@@ -21,6 +21,19 @@
    ^Session (references/get-instance session)
    ^UserInfo (references/get-instance user-info)))
 
+(defn make-proxy
+  [{:keys [type host port]}]
```
Comment (retrogradeorbit) [inline src/clj/pod/epiccastle/bbssh/pod/session.clj]: would be good to take :username and :password options, too. https://github.com/mwiede/jsch/blob/b3e4481583934e5e3ed40ce7bd43d3c4f95e6cda/src/main/java/com/jcraft/jsch/ProxySOCKS5.java#L69

File: src/bb/pod/epiccastle/bbssh/core.clj [hunk 2]
```diff
@@ -246,6 +246,7 @@
   - `:host-key-repository` Use a custom host-key-repository in the
     connection. Specifying this value will override `:known-hosts`
     setting (your host-key-repository functions will be called instead).
+  - `:proxy` Use a http or socks proxy to make the connection
```
Comment (retrogradeorbit) [inline src/bb/pod/epiccastle/bbssh/core.clj]: right here it should also document what needs to be passed to :proxy

File: src/bb/pod/epiccastle/bbssh/session.clj [hunk 3]
```diff
@@ -170,3 +170,12 @@
   (session/set-host-key-repository
    (cleaner/split-key session)
    (cleaner/split-key host-key-repository)))
+
+(defn set-proxy
+  "sets the http/socks proxy to connect with the ssh server."
```
Comment (retrogradeorbit) [inline src/bb/pod/epiccastle/bbssh/session.clj]: docstring should document what needs to be passed to it

**Review by retrogradeorbit — COMMENTED** [review]
looking good.

**Comment by lins05** [comment]
@retrogradeorbit Just addressed your comments, and added integration tests for this feature.

File: test/Dockerfile [hunk 4]
```diff
@@ -5,6 +5,13 @@ RUN apk add openssh-server openssh file curl
...
+# much as possible across tests.
+RUN wget -O /tmp/gost.gz https://github.com/ginuerzh/gost/releases/download/v2.11.5/gost-linux-amd64-2.11.5.gz \
+    && gunzip -c -k /tmp/gost.gz > /usr/local/bin/gost \
+    && chmod +x /usr/local/bin/gost \
+    && rm -f /tmp/gost.gz
 ENTRYPOINT ["/usr/sbin/sshd","-D","-E","/tmp/sshd.log"]
-EXPOSE 22/tcp
\ No newline at end of file
+EXPOSE 22/tcp
+RUN echo "root:$root_password"|chpasswd
```
Comment (lins05) [inline test/Dockerfile:17]: This line is moved to the bottom of the Dockerfile so all the previous layers (including "EXPOSE" and "ENTRYPOINT") could be cached even when `root_password` arg is changed.

**Comment by retrogradeorbit** [comment]
Really great stuff. Thanks for the PR!

**Pull request merged**